.github/workflows/build_and_run_coap_examples.yml

```diff
@@ -0,0 +1,71 @@
+name: Build and Run CoAP Examples
```

Comment:
All examples should now be build&run by this GH action https://github.com/espressif/idf-extra-components/blob/master/.github/workflows/build_and_run_examples.yml

Could you please rebase your PR and check that it works OK with your new examples?

Reply:
@tore-espressif This example runs on ethernet based runner (tag `ESP32-ETHERNET-KIT`). Is it possible to specify specific tag for an example in the `idf-build-apps` action?

Reply:
Also I don't think it will build coap_server example as it does not have a pytest script. For `coap_server`, we just want to build it in CI and it does not need to run in CI.

Reply:
@hmalpani the CI script will build all examples and run example tests for examples that have pytest

@mahavirj I'll check the tags

Reply:
Hello @mahavirj , if you want to run some example's tests on specific runner, you need to do the following:
1. Define new env marker in pytest.ini (e.g. `ethernet`)
2. Add the marker to your pytests, e.g. `@pytest.mark.ethernet`
3. Define new CI step that [runs-on](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idruns-on) a selected runner (e.g. `runs-on: [self-hosted, linux, docker, "${{ matrix.idf_target }}", "some_other_marker"]`)
4. Run `pytest -m some_other_marker` to run only testcases with this marker

Reply:
Hello @tore-espressif
I have updated `.build-test-rules.yml` file as follows: https://github.com/espressif/idf-extra-components/pull/188/files#diff-c2ae404d9e16bd33ade435e00a2de9a59245c24a72b8c24c4ef39d3ec9b59325R13-R19 so that `coap_client` and `coap_server` examples don't get built for esp32h2. But still, they are being built for esp32h2, see job: https://github.com/espressif/idf-extra-components/actions/runs/5343770280/jobs/9687302992. Am I missing something here?

Reply:
@hmalpani Shouldn't the entry be `coap/examples/coap_client`?

Reply:
@mahavirj Yes that worked. I thought we just add the example name.
Thanks!